python/ray/tune/ray_trial_executor.py

```diff
@@ -418,11 +446,18 @@ def _train(self, trial):
                             "If used with a custom stopper or early stopping, "
                             "checkpoints may be created later than desired.")
 
-            if buffer_length > 1:
+            if buffer_length > 1 and not self._is_single_process:
                 if trial.checkpoint_freq > 0:
                     buffer_length = min(buffer_length, trial.checkpoint_freq)
                 remote = trial.runner.train_buffered.remote(
                     buffer_time_s, buffer_length)
+            elif self._is_single_process:  # train happens immediately in a blocking fashion.
+                train_result = trial.runner.train()
```

Comment:
So this means we do not support buffered training here, right? That's fine, but should we raise a warning if the user specified a buffer value?

Reply:
what is buffered training? 
Can you think of anything along this line, that may make sense in normal cluster set up but not really in single process mode?